### ISSUE TITLE
multiple hosts.

### ISSUE THREAD

**Issue by solipsist01** [MikeDombo/xen-exporter#1]
Hello,

Is it possible to get metrics from multiple hosts ?
i have a cluster of 8 Citrix XenServers.
i can only run the collector on the master, and when i do, i'm only getting host and vm information of the master.

I saw in your code that you handle multiple hosts, so, i was wondering if i'm missing something obvious.

i tried connecting this exporter to the second host (which is a slave)
the docker logs from that container gives this error.

```
Traceback (most recent call last):
  File "/docker/data/prometheus/test.py", line 203, in <module>
    collect_metrics()
  File "/docker/data/prometheus/test.py", line 111, in collect_metrics
    with Xen("https://" + xen_host, xen_user, xen_password, verify_ssl) as xen:
  File "/docker/data/prometheus/test.py", line 69, in __init__
    self.session.xenapi.login_with_password(
  File "/usr/local/lib/python3.9/dist-packages/XenAPI/XenAPI.py", line 264, in __call__
    return self.__send(self.__name, args)
  File "/usr/local/lib/python3.9/dist-packages/XenAPI/XenAPI.py", line 151, in xenapi_request
    self._login(methodname, params)
  File "/usr/local/lib/python3.9/dist-packages/XenAPI/XenAPI.py", line 175, in _login
    result = _parse_result(
  File "/usr/local/lib/python3.9/dist-packages/XenAPI/XenAPI.py", line 238, in _parse_result
    raise Failure(result['ErrorDescription'])
XenAPI.XenAPI.Failure: ['HOST_IS_SLAVE', '10.10.10.131']
```

**Comment by MikeDombo** [comment]
I don't have multiple hosts in order to test this, so the multi-host support only exists in theory.

The master is supposed to have the metrics for all hosts, so running it only on the master is correct.

If you're able to figure out how to make it work with multiple hosts I'd love to accept a pull request to add that feature properly.

**Comment by solipsist01** [comment]
Hello,

Thank you for your answer.
Upon further examination, it's possible to manually download the RRD info from the slave.

It's the XenAPI module which doesn't alllow to connect to a slave.
RRD is however different. data is stored on the host itself according to the docs. (except if a VM is powered off)

Tomorrow i'm going to try to rewrite to requests, as we're downloading only a json anyway.
If i get it working, i'll do a PR.

**Comment by MikeDombo** [comment]
Merged #2.
